tests/test_sshfsmount.cpp

```diff
@@ -241,3 +247,107 @@ TEST_F(SshfsMount, unblocks_when_sftpserver_exits)
     mount.join();
     EXPECT_TRUE(stopped_ok);
 }
+
+TEST_F(SshfsMount, throws_when_unable_to_change_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"cd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, throws_when_unable_to_get_current_dir)
+{
+    bool invoked{false};
+    auto request_exec = make_exec_that_fails_for({"pwd"}, invoked);
+    REPLACE(ssh_channel_request_exec, request_exec);
+
+    EXPECT_THROW(make_sshfsmount(), std::runtime_error);
+    EXPECT_TRUE(invoked);
+}
+
+TEST_F(SshfsMount, executes_commands)
+{
+    std::unordered_map<std::string, std::string> commands = {
+        {"which sshfs", "/usr/bin/sshfs"},
+        {"pwd", "/home/ubuntu"},
+        {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+         "/home/ubuntu/"},
+        {"sudo /bin/bash -c 'cd \"/home/ubuntu/\" && mkdir -p \"target\"'", ""},
+        {"id -nu", "ubuntu"},
+        {"id -ng", "ubuntu"},
+        {"id -u", "1000"},
+        {"id -g", "1000"},
+        {"sudo sshfs -o slave -o nonempty -o transform_symlinks -o allow_other :\"source\" \"target\"", "don't care"}};
+
+    std::unordered_map<std::string, std::string> executed;
+
+    bool invoked{false};
+    std::string output{"1000"};
+    auto remaining = output.size();
```

Comment:
This seems arbitrary, shouldn't these be empty to start with?

Reply:
Sure, they were a leftover from a previous implementation.